extensions/amp-brightcove/0.1/amp-brightcove.js

```diff
@@ -34,7 +34,19 @@ class AmpBrightcove extends AMP.BaseElement {
   }
 
   /** @override */
+  buildCallback() {
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {?Promise} */
+    this.iframePromise_ = null;
+  }
+
+  /** @override */
   layoutCallback() {
+    if (this.iframePromise_) {
```

Comment:
I see. You are protecting against re-layout after unlayout which is allowed. However, this code is problematic - it actually prevents relayout-after-layout from happening.